[OSX]Touchups

RowenStipe opened the pull request:
This commit has several touch ups that only effect the OS X clients.

`info.plist` and `qt.pro` are now configured to create an automatic
short version to help distinguish between clients clearly and include a
referenced file not put into the build when compiled.
`./img/icon/qtox_profile.icns’

`info.plist` also had it’s bundle version to be  updated to the 1.2.2
tag and supported languages added and fixed referenced file.

The default save location has also been moved to comply with the Tox
Client Standard. (Previous location was `~/Library/Prefrences/tox` )

A basic bash profile mover has also been included to show I did think
of that.

RowenStipe committed:
Touchups

This commit has several touch ups that only effect the OS X clients.

`info.plist` and `qt.pro` are now configured to create an automatic
short version to help distinguish between clients clearly and include a
referenced file not put into the build when compiled.
`./img/icon/qtox_profile.icns’

`info.plist` also had it’s bundle version to be  updated to the 1.2.2
tag and supported languages added and fixed referenced file.

The default save location has also been moved to comply with the Tox
Client Standard. (Previous location was ~/Library/Prefrences/tox )

A basic bash profile mover has also been included to show I did think
of that.

zetok commented on osx/info.plist:
```diff
@@ -9,7 +9,7 @@
 	<string>APPL</string>
 	<key>CFBundleSignature</key>
-	<string>????</string>
+        <string>toxq</string>
```
Shouldn't tabs be used instead of spaces?

tux3 commented on src/persistence/settings.cpp:
```diff
@@ -537,6 +537,9 @@ QString Settings::getSettingsDirPath()
                            + "AppData" + QDir::separator() + "Roaming" + QDir::separator() + "tox")+QDir::separator();
+#elif defined(Q_OS_OSX)
+    return QDir::cleanPath(QStandardPaths::writableLocation(QStandardPaths::HomeLocation) + QDir::separator()
+                           + "Library" + QDir::separator() + "Application Support" + QDir::separator() + "Tox")+QDir::separator();
```
If you change the settings dir path without writing compatibility code, people will lose their profiles when they update!

RowenStipe committed:
Performed auto-indentation

Ran an auto indentation script included in one of my text editors
(TextMate:greenheart:)

RowenStipe committed:
Added basic migration compatibility code.

This code works if the ~/Library/Application Support/Tox directory does
not already exist.

RowenStipe commented:
NOTE: This pull is still a work in progress. I want to get the profile migration code to work even if the new directory exists.

tux3 commented on src/platform/install_osx.cpp:
```diff
@@ -82,3 +84,27 @@ void osx::moveToAppFolder()
     }
 }
+
+void osx::migrateProfiles()
```
Also, you could leave a comment saying that this is temporary migration code, so that when it gets forgotten about 10 years later we'll know it can be safely removed :)

RowenStipe committed:
Added case for `/Application Support/Tox` already existing

I tried to get it done without bash… But this was the simplest method
that works for this special case scenerio.

RowenStipe commented:
Going to see if I can add: #2814 into this pr as well.

RowenStipe committed:
Enable OS X Auto-run at login

RowenStipe commented:
Unless there's something major I'm not seeing that needs to be fixed. This PR is ready.
